www/source/partials/docs/_dev-pkgs-share.html.md.erb

```diff
@@ -31,23 +31,23 @@ When you upload a package to Builder, you are required to supply an auth token a
 
-### Create origin keys
+### Create an origin key pair
 
-After you have done the basic account creation steps, you need to create your origin keys. The private key will be used to sign your packages and the public key will be used by supervisors to verify the integrity of your packages (`.hart` files).
+After finishing the basic account creation steps, you need to create your origin key pair. You will use the private key to sign your packages and Supervisors will use the public key to verify the integrity of your packages (`.hart` files).
 
-You can either create an origin key pair by running `hab cli setup` from your host machine, or running `hab origin key generate <ORIGIN>` from either the host machine or from within the studio.
+You can create an origin key pair by running `hab cli setup` from your host machine, or by running `hab origin key generate <ORIGIN>` from either the host machine or from within the studio.
 
-Your keys are located at `~/.hab/cache/keys` on your host machine and `/hab/cache/keys` inside the studio environment.
+Your public and private origin keys are located at `~/.hab/cache/keys` on your host machine and at `/hab/cache/keys` inside the studio environment.
 
 ### Upload the keys for your origin
 
-If you created a new origin and/or Builder does not have the public key that corresponds to the private key used to build your package, you must upload it. You also have the ability to upload your private key; however, if you do not upload at least the public key, Builder will reject the upload of your packages for that origin.
+If you created a new origin and/or Builder does not have the public key corresponding to the private key used to build your package, you must upload it. You also have the ability to upload your private key; however, if you do not upload at least the public key, Builder will reject the upload of your packages for that origin.
 
 You can upload keys for the origin through the web interface for Builder, or by using the `hab origin key upload` command. You must be authenticated using the access token described earlier before you can upload keys.
 
 ### Upload packages to Builder
 
-Once Builder possesses at least the public key of the origin, you may upload one or more packages to that origin by using the `hab pkg upload` command. Builder will check the cryptographic integrity of the package before allowing you to upload it. Uploading packages is also a privileged operation for which you must have the access token.
+Once Builder possesses at least the public origin key, then you may upload one or more packages to that origin with the `hab pkg upload` command. Builder will check the cryptographic integrity of the package before allowing you to upload it. Uploading packages is also a privileged operation for which you must have the access token.
```

Comment:
> Builder will check the cryptographic integrity of the package before allowing you to upload it.

Is kind of a weird thing to say because builder can't **do** any verification until the artifact is uploaded. It may decline to *store* an artifact that fails verification, but the way it's phrased now is misleading.

Reply:
Agree. I'm having trouble nailing down the who/what/when of key verification and use.

Reply:
I think @chefsalim is the subject matter expert on what builder does with respect to upload and verification.